### PR TITLE
feat(firebase): new autoDispose option for useFirestore & useRTDB.

### PR DESCRIPTION
**Pull request by mmommo**
Hi Ant,

First time for PR and not really familiar with TS. Haven't tested locally as lots of configurations required to run on my local machine. Still learning.

Cheers!

### CODE REVIEW

File: packages/firebase/useFirestore/index.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ function isDocumentReference<T>(docRef: any): docRef is firebase.firestore.Docum
 
 export interface FirestoreOptions {
   errorHandler?: (err: Error) => void
+  autoDispose? : true
```
Comment (antfu) [inline packages/firebase/useFirestore/index.ts:30]:
```suggestion
  autoDispose?: boolean
```

File: packages/firebase/useRTDB/index.ts [hunk 2]
```diff
@@ -2,14 +2,19 @@ import type firebase from 'firebase'
 import { Ref, ref } from 'vue-demi'
 import { tryOnUnmounted } from '@vueuse/shared'
 
+export interface RTDBOptions {
+  autoDispose? : true
```
Comment (antfu) [inline packages/firebase/useRTDB/index.ts:6]:
```suggestion
  autoDispose?: boolean
```

File: packages/firebase/useRTDB/index.ts [hunk 3]
```diff
@@ -19,9 +24,10 @@ export function useRTDB<T = any>(
 
   docRef.on('value', update)
 
-  tryOnUnmounted(() => {
-    docRef.off('value', update)
-  })
-
+  if (options.autoDispose) {
```
Comment (antfu) [inline packages/firebase/useRTDB/index.ts:27]: Will need to do the object destucturing like `useFirebase` does to assign the default value.

**Review by antfu — COMMENTED** [review]
Looks good overall, it would be great if you can update the docs reflecting this, thanks!

File: packages/firebase/useRTDB/index.ts (hunk 3 above)
Comment (mmommo) [reply packages/firebase/useRTDB/index.ts]: done with updates.

**Comment by mmommo** [comment]
And how do I change the doc?

**Comment by antfu** [comment]
The `index.md` in the same dir

**Comment by mmommo** [comment]
updated doc for useFirestore. not sure if we still need to keep createGlobalState as autoDispose has served similar purpose. Please comment. Cheers!

**Comment by antfu** [comment]
@mmommo thanks, I think we can still keep `createGlobalState` offering more options for users' preference. Can you fix the linter? Thanks.

**Review by antfu — CHANGES_REQUESTED** [review]
Could you revert the removal of "Share across instances"? Thanks.

**Review by antfu — APPROVED** [review]

**Pull request merged**